### PR TITLE
Adds fetch, caching, and background refresh of metrics files from GCS

### PR DESCRIPTION
**Pull request by jessex**
This goes towards, but does not complete, #28. This includes:
* API endpoints on the backend for fetch admission, reincarceration, and revocation metrics
* Fetching of metrics from pre-processed json files in GCS, organized by general metric variety (the 3 aforementioned types)
* Caching of those metrics in memory so that each call does not round-trip to GCS
* Background refresh of that cache so that only rare calls actually invoke a round-trip to GCS
* Authentication against GCS via service accounts, whose credentials are stored in secret files referenced by environment variables

This does not yet hook up these metrics to any visualizations.

Tested with lots of local manual testing. Requests from the browser to the API when the metrics are in the cache lead to a cached 304 response in the sub-millisecond range. The logs show the background refresh executing on schedule, hitting GCS to refresh the cache only after the cache TTL has expired.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -27,6 +27,8 @@ auth_config.json
 auth_config*.json
 gae*.yaml
 src/assets/scripts/metrics/*.json
+recidiviz-production-*.json
+recidiviz-staging-*.json
```
Comment (jjdressel11) [inline .gitignore]: Suggestion to change these files to start with `recidiviz-dev-` instead so we don't have the name of our GCP project in these files?

File: package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "test": "react-scripts test",
     "spa": "react-scripts start",
     "server": "node server.js",
-    "server:dev": "AUTH_ENV=development nodemon server.js",
+    "server:dev": "AUTH_ENV=development GOOGLE_APPLICATION_CREDENTIALS=./recidiviz-staging-service-account.json nodemon server.js",
```
Comment (jjdressel11) [inline package.json]: Per my previous comment, this would be `recidiviz-dev-service-account.json`

File: server/core/metricsApi.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+ * In the current implementation, metrics are stored in pre-processed json files in Google Cloud
+ * Storage. Those files are pulled down and cached in memory with a TTL. That TTL is unaffected by
+ * access to the cache, so files are re-fetched at a predictable cadence, allowing for updates to
+ * those files to be quickly reflected in the app without frequent requests to GCS.
+ */
+
+var objectStorage = require('./objectStorage');
+var cacheManager = require('cache-manager');
+
+const BUCKET_NAME = 'recidiviz-staging-dashboard-data';
```
Comment (jjdressel11) [inline server/core/metricsApi.js]: Is this a temporary solution until we are accessing the bucket in prod? Or will we have the prod bucket in plain text in prod?

File: .gitignore (hunk 1 above)
Comment (jessex) [reply .gitignore]: Good callout, I'll change that.

File: package.json (hunk 2 above)
Comment (jessex) [reply package.json]: Will change!

File: server/core/metricsApi.js (hunk 3 above)
Comment (jessex) [reply server/core/metricsApi.js]: Ah you're right, I meant to extract that to an environment variable before submitting the PR. I'll hide this behind an environment variable, and we can rename the bucket if we'd like for security's sake.

**Pull request merged**